vhdl_parser/src/declarative_part.rs

```diff
@@ -82,46 +83,69 @@ pub fn parse_declarative_part_leave_end_token(
     messages: &mut dyn MessageHandler,
 ) -> ParseResult<Vec<Declaration>> {
     let mut declarations: Vec<Declaration> = Vec::new();
+    let multi_decl_tokens = &[File, Shared, Constant, Signal, Variable, Attribute];
+    let rec_tokens = &[
+        SemiColon, Begin, End, Use, Type, Subtype, Component, Alias, Impure, Function, Procedure,
+        Package, For, File, Shared, Constant, Signal, Variable, Attribute,
+    ];
 
     while let Some(token) = stream.peek()? {
         match token.kind {
-            Use => declarations.push(Declaration::Use(parse_use_clause(stream)?)),
-            Type | Subtype => {
-                declarations.push(Declaration::Type(parse_type_declaration(stream, messages)?))
-            }
-            Shared | Constant | Signal | Variable => {
-                for decl in parse_object_declaration(stream)? {
-                    declarations.push(Declaration::Object(decl))
-                }
-            }
-            File => {
-                for decl in parse_file_declaration(stream)? {
-                    declarations.push(Declaration::File(decl))
+            Begin | End => break,
+
+            kind if multi_decl_tokens.contains(&kind) => {
```

Comment:
Using Vec.contains is O(n) which is not ideal. It is better to create an `is_<xyz>_kinds` function that uses a match statement internally to return true/false